tests/unittest.py

```diff
@@ -33,9 +37,8 @@
 USE_MEMORY_STORAGE = os.environ.get('HATHOR_TEST_MEMORY_STORAGE', 'false').lower() == 'true'
 
 
-def shorten_hash(container):
-    container_type = type(container)
-    return container_type(h[-2:].hex() for h in container)
+def shorten_hash(container: Collection[bytes]) -> Iterable[str]:
+    return map(lambda hash_bytes: hash_bytes[-2:].hex(), container)
```

Comment:
I guess this new code does something different then the original code.

Reply:
It is implemented differently, but the result should be the same. I changed it because it was hard to correctly type it with the previous implementation, and this function is only ever used in debug logs.

Reply:
Can we rename this to `short_hashes`? The signature makes it more clear that it takes a list/container instead of a single bytes instance, which is what I thought it did from the function name.

Reply:
Done in https://github.com/HathorNetwork/hathor-core/pull/974/commits/6ff78af167152e8f0737a5f46f6096ccc0a6c419

Reply:
I agree there's a slight difference in meaning, the original would return a `set` when the input is a `set`. Typing wise the original behavior could be described with generics:

```python
C = TypeVar('C', bound=Collection)
def short_hashes(container: C[bytes]) -> C[str]
```

But I'm OK either way.

Reply:
@jansegre I don't think Python supports type arguments on type vars, at least not like this. I wish this was Rust 😂

Reply:
I see, apparently it's a really complex feature from mypy's POV: https://github.com/python/typing/issues/548

Reply:
Ok. Thanks!